tests/e2e/vsphere.go

```diff
@@ -1137,3 +1137,16 @@ func fetchDsUrl4CnsVol(e2eVSphere vSphere, volHandle string) string {
 	gomega.Expect(queryResult.Volumes).ShouldNot(gomega.BeEmpty())
 	return queryResult.Volumes[0].DatastoreUrl
 }
+
```

Comment:
verifyDatastoreMatch is not returning any bool value. Immediately it is failing if datastore match is not found.
In few of my testcases I have given only 1 preferred shared datastore but allowed topology is set to all racks/clusters.
So volume can be provisioned on any of the cluster/racks(means any of the datastore along with preferred datastore)
If volume is not provisioned on the preferred datastore, the code was failing immediately if no datastore match found but volume provision can happen on any of the datastore because multiple allowed topology is set.
hence i have written my own wrapper method where I am not failing the code rather returning the bool value and later in the logic i m checking the datastore preference count